papers/opencadd-klifs/paper.md

```diff
@@ -0,0 +1,66 @@
+
+# Statement of need
+
+<!---
+A Statement of Need section that clearly illustrates the research purpose of the software.
+-->
+
+OpenCADD is a collection of Python modules for structural cheminformatics; OpenCADD-KLIFS is one of them (``opencadd.databases.klifs``).
+This module offers access to KLIFS data such as information about kinases, structures, ligands, 
+interaction fingerprints, and bioactivities. 
+With OpenCADD-KLIFS, KLIFS data can be queried either locally from a KLIFS download or remotely from the KLIFS webserver. 
+The tool provides identical APIs for the remote and local queries and streamlines all output into 
+standardized Pandas [@pandas] DataFrames to allow for an easy and quick downstream data manipulation (Figure \autoref{fig:opencadd_klifs_toc}). This Pandas-focused setup is ideal to work with Jupyter notebooks [@jupyterhub; @Kluyver:2016]. 
+
+
+![OpenCADD-KLIFS fetches KLIFS [@Kanev:2021] data offline from a KLIFS download or online from the KLIFS database and formats the output in user-friendly Pandas [@pandas] DataFrames.\label{fig:opencadd_klifs_toc}](opencadd_klifs_toc.png)
+
+The KLIFS database offers a REST API including an OpenAPI specification. OpenCADD-KLIFS uses bravado [@bravado] to dynamically generate a Python client based on the OpenAPI definitions and add wrappers to enable the following functionalities:
+
+- A session is set up, which allows access to various KLIFS data sources by different identifiers with the API ``session.data_source.by_identifier``; for example ``session.structures.by_kinase_name`` fetches information on all structures for a query kinase. Data sources currently include kinases, ligands, structures, drugs, pockets, bioactivities, structural conformations, modified residues, and coordinates; identifiers refer to kinase names, PDB IDs, KLIFS IDs, and more.
+- Query results obtained from the remote KLIFS webserver are streamlined with those obtained from a local KLIFS download using the same API.
+- All results obtained with bravado are formatted as Pandas DataFrames with standardized column names, data types, and handling of missing data.
+- Structural files deposited on KLIFS include full complexes or selections such as proteins, pockets, ligands, and more. These files can be downloaded to disc or loaded via biopandas [@biopandas] or the RDKit [@rdkit].
+
+OpenCADD-KLIFS is especially useful whenever users are interested in multiple or more complex queries such as "fetch all interaction profiles of kinases bound to the drug Gefitinib". Formatting the output as DataFrames facilitates subsequent filtering steps and DataFrame merges in case multiple KLIFS datasets need to be combined.
+OpenCADD-KLIFS is currently used in several projects from the Volkamer Lab [@volkamerlab] including TeachOpenCADD [@teachopencadd], OpenCADD-pocket [@opencaddpocket], KiSSim [@kissim], KinoML [@kinoml], and PLIPify [@plifify].
+
```

Comment:
Yes, I like the new example!

Now since the extended example is already about 'fetching interaction profiles', maybe we can exchange the more general example in the beginning of the section, so show a different aspect as well?
`OpenCADD-KLIFS is especially convenient whenever users are interested in multiple or more complex queries such as "fetching all interaction profiles of kinases bound to the drug Gefitinib". `

Reply:
e.g. 
* ... such as "fetching all kinase complexes bound to the drug Gefitinib". 
* or ... such as "fetching all structures for kinase EGFR in DFG-in conformation".

Reply:
Good idea! I settled for 

> OpenCADD-KLIFS is especially convenient whenever users are interested in multiple or more complex queries such as "fetching all structures for the kinase EGFR in the DFG-in conformation" or "fetching the measured bioactivity profiles for all ligands that are structurally resolved in complex with EGFR".

This example

> fetching all kinase structures bound to the drug Gefitinib

requires only one query thanks to the KLIFS API (at least when you know the Expo ID for Gefitinib).